core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -110,6 +117,14 @@ class HistoryServer(
     appCache.getSparkUI(appKey)
   }
 
+  val historyMetrics = new HistoryMetrics(this, "history.server")
+
+  /**
+   * Provider metrics are None until the provider is started, and only after that
+   * point if the provider returns any.
+   */
+  var providerMetrics: Option[Source] = None
```

Comment:
not easily, would be the summary. making it a `private[history]` would lock down access though.